firrtl/src/main/scala/firrtl/graph/DiGraph.scala

```diff
@@ -279,11 +279,14 @@ class DiGraph[T](private[graph] val edges: LinkedHashMap[T, LinkedHashSet[T]]) {
         if (frame.childCall.isEmpty) {
           if (lowlinks(v) == indices(v)) {
             val scc = new mutable.ArrayBuffer[T]
-            do {
+            val w = stack.pop()
+            onstack -= w
+            scc += w
+            while (scc.last != v) {
               val w = stack.pop()
               onstack -= w
               scc += w
-            } while (scc.last != v);
+            }
```

Comment:
Isn't the suggested transformation to just make the look the condition?
```scala
while({
  val w = stack.pop()
  onstack -= w
  scc += w
  scc.last != v
}) ()
// or you can end it more explicitly with
}) { /* empty body, logic is in condition */ }
```

Reply:
Is DiGraph still using? How about directly remove it?

Reply:
It's used by the `DependencyManager` for computing phase orderings. It can't go just yet. However, it could be made private if that is useful.

Reply:
oh I forgot this. They should go away eventually.